tl

```diff
@@ -273,6 +245,69 @@ if not cmd then
    os.exit(1)
 end
 
+local function get_config()
+   local config = {
+      preload_modules = {},
+      include_dir = {},
+      disable_warnings = {},
+      warning_error = {},
+      quiet = false
+   }
+
+   local config_path = cmd == "build"
+      and find_file_in_parent_dirs("tlconfig.lua")
+      or "tlconfig.lua"
+
+   local conf, err = loadfile(config_path)
+   if not conf then
+      if err:match("No such file or directory$") then
+         if cmd == "build" then
+            die("Build error: tlconfig.lua not found")
+         end
+      else
+         die("Error loading config: " .. err)
+      end
+   end
+
+   if conf then
+      local ok, user_config = pcall(conf)
+      if not ok then
+         die("Error loading config:" .. err)
+      end
+
+      -- Merge tlconfig with the default config
+      if user_config then
+         for k, v in pairs(user_config) do
+            config[k] = v
+         end
+      end
+   end
+
+   if not config.build_file then
+      if lfs.attributes("./build.tl", "mode") == "file" then
```

Comment:
the lfs.chdir below is happening after this check, right? It seems like this check/adjustment of the `build_file` value should be moved below it (or the lfs.chdir should be moved in here).

Reply:
Yep, good catch